### PR TITLE
Add report details

### PR DESCRIPTION
**Pull request by sven-oly**
Latest updates supporting more detail in reports.

### CODE REVIEW

File: verifier/testreport.py [hunk 1]
```diff
@@ -198,29 +139,57 @@ def summary_status(self):
     return self.tests_fail == 0 and not self.missing_verify_data
 
   def compute_category_summary(self, items, group_tag, detail_tag):
```
Comment (echeran) [inline verifier/testreport.py:141]: Rename `items` parameter to `test_errors`, or maybe something more generic if it's errors + unsupported exceptions + failures...
Rename function name to have the word "error" in it to be more descriptive/precise.

File: verifier/testreport.py [hunk 2]
```diff
@@ -198,29 +139,57 @@ def summary_status(self):
...
+        if groups.get(group):
+          groups[group][str(detail)].append(label)
         else:
-          groups[group]= [item]
-    print('GROUPS FOR %s %s = \n%s' % (group_tag, detail_tag, groups.keys()))
+          groups[group]= {str(detail): [label]}
+    # print('GROUPS FOR %s %s = \n%s' % (group_tag, detail_tag, groups.keys()))
+    return groups
+
+  def compute_unsupported_category_summary(self, items, group_tag, detail_tag):
```
Comment (echeran) [inline verifier/testreport.py:165]: Ditto on renaming suggestions.

File: verifier/testreport.py [hunk 3]
```diff
@@ -198,29 +139,57 @@ def summary_status(self):
...
+    # For the items, count messages and arguments for each
+    groups = {}
+    for item in items:
+      error_detail = item.get('error_detail')
```
Comment (echeran) [inline verifier/testreport.py:169]: Rename local variable identifiers.
Also consider renaming key in test executor result structs from `error_detail` to something more generic if errors and unsupported exceptions can both occur.

File: verifier/testreport.py [hunk 4]
```diff
@@ -332,6 +341,58 @@ def create_html_report(self):
...
+      # Sort these by number of items in each set.
+
+      # Find the largest intersections of these sets
+      combo_list = [ (combo, len(results[combo])) for combo in results]  #.sort(key=takeSecond)
+      # sort combo_list by size of sets
+      combo_list.sort(key=takeSecond, reverse=True)
+
+    return results
+
+  def characterizeFailuresByTestType(self):
```
Comment (echeran) [inline verifier/testreport.py:380]: Remove this if no longer used.

File: verifier/testreport.py [hunk 5]
```diff
@@ -332,6 +341,58 @@ def create_html_report(self):
     file.close()
     return html_output
 
+  def setsize(s):
```
Comment (echeran) [inline verifier/testreport.py:344]: Rename? Remove if not used?

File: verifier/testreport.py [hunk 6]
```diff
@@ -332,6 +341,58 @@ def create_html_report(self):
...
+      # Get input_data, if available
+      label = test['label']
+      if test.get('input_data'):
+        # Look at locale
+        input_data = test.get('input_data')
+
+        if input_data.get('locale'):
+          failure_combo = 'locale' + ':' + input_data.get('locale')
+          if failure_combo not in results:
+            results[failure_combo] = set()
```
Comment (echeran) [inline verifier/testreport.py:361]: Consider [`defaultdict`](https://docs.python.org/3/library/collections.html#collections.defaultdict) for this type of thing.

File: verifier/testreport.py [hunk 7]
```diff
@@ -447,66 +508,8 @@ def find_replacements_diff(self, s1, s2):
...
-    self.header_item_template = Template(
-        '<th>$header_data</th>'
-        )
-    self.line_template = Template(
-        '<tr>$column_data</tr>'
-        )
-    self.entry_template = Template(
-        '<td>$report_detail</td>'
-        )
+def takeSecond(elem):
```
Comment (echeran) [inline verifier/testreport.py:511]: Needed?

File: verifier/testreport.py [hunk 8]
```diff
@@ -311,14 +304,30 @@ def create_html_report(self):
...
+      unsupported_table = self.templates.summary_table_template.safe_substitute(
+          {'table_content': ('\n').join(unsupported_summary_lines),
+           'type': 'Unsupported options'}
+      )
+
+      html_map['unsupported_summary'] = unsupported_table
     else:
       html_map['unsupported_section'] = 'No unsupported tests found'
+      html_map['unsupported_summary'] = ''
+    html_map['unsupported_labels'] = unsupported_summary
```
Comment (echeran) [inline verifier/testreport.py:328]: https://github.com/unicode-org/conformance/actions/runs/4199930034/jobs/7285402355
```
Traceback (most recent call last):
  File "/home/runner/work/conformance/conformance/verifier/verifier.py", line 506, in <module>
    main(sys.argv)
  File "/home/runner/work/conformance/conformance/verifier/verifier.py", line 497, in main
    verifier.verifyDataResults()
  File "/home/runner/work/conformance/conformance/verifier/verifier.py", line 196, in verifyDataResults
    self.report.create_html_report()
  File "/home/runner/work/conformance/conformance/verifier/testreport.py", line 328, in create_html_report
    html_map['unsupported_labels'] = unsupported_summary
UnboundLocalError: local variable 'unsupported_summary' referenced before assignment
```

**Review by sven-oly — COMMENTED** [review]
Verifier has a failure with nodejs 19.7.

**Comment by sven-oly** [comment]
Merging to get this into our code base.

**Pull request merged**